pkg/policy/policy.go

```diff
@@ -26,6 +27,18 @@ import (
 	"github.com/op/go-logging"
 )
 
+// Completion is an interface for an individual asynchronous success/failure
+// notification.
+type Completion interface {
+	Completed(bool)
+}
+
+// CompletionContainer is an interface for pooling asynchronous success/failure
```

Comment:
pooling -> polling (?)

Reply:
Pooling is the right spelling here :-)

Multiple completions can be added to the same container, so that the caller can efficiently wait for multiple compoletions at one time.